sqlx-core/src/postgres/types/money.rs

```diff
@@ -0,0 +1,217 @@
+use crate::{
+    decode::Decode,
+    encode::{Encode, IsNull},
+    error::BoxDynError,
+    postgres::{PgArgumentBuffer, PgTypeInfo, PgValueFormat, PgValueRef, Postgres},
+    types::Type,
+};
+use byteorder::{BigEndian, ByteOrder};
+use std::{
+    io,
+    ops::{Add, AddAssign, Sub, SubAssign},
+};
+
+/// The PostgreSQL [`MONEY`] type stores a currency amount with a fixed fractional
+/// precision. The fractional precision is determined by the database's
+/// `lc_monetary` setting.
+///
+/// Data is read and written as 64-bit signed integers, and conversion into a
+/// decimal should be done using the right precision.
+///
+/// Reading `MONEY` value in text format is not supported and will cause an error.
+///
+/// [`MONEY`]: https://www.postgresql.org/docs/current/datatype-money.html
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct PgMoney(pub i64);
+
+impl PgMoney {
+    /// Convert the money value into a [`BigDecimal`] using the correct precision
+    /// defined in the PostgreSQL settings. The default precision is two.
+    ///
+    /// [`BigDecimal`]: ../../types/struct.BigDecimal.html
+    #[cfg(feature = "bigdecimal")]
+    pub fn as_bigdecimal(&self, scale: i64) -> bigdecimal::BigDecimal {
```

Comment:
PR looks good. My only nit is it probably makes more sense to `impl From<PgMoney> for BigDecimal` as its infallible and `PgMoney` should probably be a `Copy` type anyway.

Reply:
How would you set the scale in this case? `impl From<(PgMoney, i64)> for BigDecimal`?